cogs/help.py

```diff
@@ -0,0 +1,238 @@
+            bot_id: int,
+            cogs: dict) -> None:
+        """Create the menu."""
+        self.get_command_signature = signature
+        self.filter_commands = filter_commands
+        self.prefix = prefix
+        self.menu_author = author
+        self.bot_id = bot_id
+        super().__init__(
+            [(cog, cogs[cog]) for cog in sorted(
+                cogs,
+                key=lambda cog: cog.qualified_name if cog else "ZZ"
+            )],
+            per_page=1,
+        )
+
+    async def format_page(
+            self,
+            menu: menus.Menu,
+            cog_tuple: tuple) -> discord.Embed:
+        """Format the pages."""
+        cog, command_list = cog_tuple
+        embed = discord.Embed(
+            title=textwrap.dedent(
+                f"""
+                Help for
+                {cog.qualified_name if cog else 'unclassified commands'}
+                """
+            ),
+            description=textwrap.dedent(
+                f"""
+                Help syntax : `<Required argument>`. `[Optional argument]`
+                Command prefix: `{self.prefix}`
+                {cog.description if cog else ""}
+                """
+            ),
+            color=0xffff00,
+        )
+        embed.set_author(
+            name=self.menu_author.display_name,
+            icon_url=str(self.menu_author.avatar_url),
+        )
+        for command in await self.filter_commands(command_list):
+            embed.add_field(
+                name=f"{self.prefix}{self.get_command_signature(command)}",
+                value=command.help,
+                inline=False,
+            )
+        embed.set_footer(
+            text=f"Page {menu.current_page+1}/{self.get_max_pages()}"
+        )
+        return embed
+
+
+class Help(commands.HelpCommand):
+    """The Help cog."""
+
+    def get_command_signature(self, command: commands.Command) -> str:
+        """Retrieve the command's signature."""
+        basis = f"{command.qualified_name}"
+        for arg in command.clean_params.values():
+            if arg.kind in (Parameter.VAR_KEYWORD, Parameter.VAR_POSITIONAL):
+                basis += f" [{arg.name}]"
+            elif arg.annotation == Optional:
+                basis += f" [{arg.name} = None]"
+            elif isinstance(arg.annotation, commands.converter._Greedy):
+                basis += f" [{arg.name} = (...)]"
+            elif arg.default == Parameter.empty:
+                basis += f" <{arg.name}>"
+            else:
+                basis += f" [{arg.name} = {arg.default}]"
+        return basis
+
+    async def send_bot_help(self, mapping: dict) -> None:
+        """Send the global help."""
+        ctx = self.context
+        pages = menus.MenuPages(
+            source=HelpSource(
+                self.get_command_signature,
+                self.filter_commands,
+                ctx.prefix,
+                ctx.author,
+                ctx.bot.user.id,
+                mapping),
```

Comment:
This ending bracket should also be unindented, ending brackets of both all functions should be on the same level as their start
```suggestion
                mapping
            ),
```

Reply:
Done it